### PR TITLE
Refactor messaging

### PR DESCRIPTION
**Pull request by angaither**
#### What's this PR do?
Still has all the same functionality as #164, but now with some updates.
Created an `Email` facade class to prepare/replace the subject
Only pass one var in, instead of so so many

#### How should this be manually tested?
Does everything still work?

#### What are the relevant tickets?
Refs #152

### CODE REVIEW

File: app/Events/QueueMessageRequest.php [hunk 1]
```diff
@@ -15,9 +15,7 @@ class QueueMessageRequest extends Event
      *
      * @var \Gladiator\Models\Message
```
Comment (weerd) [inline app/Events/QueueMessageRequest.php]: does the DocBlock still apply or change to like `@var array` or something?

File: app/Events/QueueMessageRequest.php [hunk 2]
```diff
@@ -27,11 +25,9 @@ class QueueMessageRequest extends Event
      *
      * @return void
      */
-    public function __construct(Message $message, $sender, Competition $competition)
+    public function __construct($email)
```
Comment (weerd) [inline app/Events/QueueMessageRequest.php]: much nicer :dancer:

File: app/Http/Controllers/CompetitionsController.php [hunk 3]
```diff
@@ -118,7 +118,7 @@ public function removeUser(Competition $competition, User $user)
     }
 
     /**
-     * Detach a user from a competition.
+     * Show and send and messages in a competition.
```
Comment (weerd) [inline app/Http/Controllers/CompetitionsController.php]: typo

File: app/Http/Controllers/MessagesController.php [hunk 4]
```diff
@@ -78,15 +79,17 @@ public function show($id)
...
+        $competitionId = request('competition_id');
 
-        $from = Contest::find($contest_id)->sender;
-        $competition = Competition::find($competition_id);
+        $email = new Email();
+        $email->message = $message;
+        $email->sender = Contest::find($contestId)->sender;
+        $email->competition = Competition::find($competitionId);
 
         //@TODO: only pass one thing in here instead of 3 or more.
```
Comment (sbsmith86) [inline app/Http/Controllers/MessagesController.php]: You can remove this TODO now!

**Comment by sbsmith86** [comment]
I definitely like this better! looking good.

File: app/Http/Controllers/MessagesController.php [hunk 5]
```diff
@@ -78,15 +79,17 @@ public function show($id)
...
-        event(new QueueMessageRequest($message, $from, $competition));
+        event(new QueueMessageRequest($email));
 
-        return redirect()->route('competitions.message', ['competition' => $competition_id, 'contest' => $contest_id])->with('status', 'Fired that right the hell off!');
+        return redirect()->route('competitions.message', ['competition' => $competitionId, 'contest' => $contestId])->with('status', 'Fired that right the hell off!');
```
Comment (weerd) [inline app/Http/Controllers/MessagesController.php]: > 'Fired that right the hell off!'

haha 🚀

File: app/Listeners/QueueMessage.php [hunk 6]
```diff
@@ -29,20 +30,19 @@ public function __construct(Mailer $mail)
      */
     public function handle(QueueMessageRequest $event)
     {
-        $content = Message::prepareMessage($event->message, $event->competition);
+        $email = $event->email;
+        $type = $email->message->type;
```
Comment (weerd) [inline app/Listeners/QueueMessage.php]: Do you need to include the `$key` as well.... so for like `reminder` types, there's like 5 of those...

Comment (weerd) [inline app/Listeners/QueueMessage.php]: Oh wait, this is for helping to specify the blade template. Nevermind.

**Comment by weerd** [comment]
Nice work, this is looking cooler :dancers:

**Comment by angaither** [comment]
@weerd @sbsmith86 updated!

**Comment by weerd** [comment]
:+1:

**Pull request merged**